File: Bedrock Server Pilot/BSPChat Addon/http_server.py
import http.server
import os

message_log_file = "message_log.txt"  # File to save the messages
latest_message = ""  # Shared variable for the latest message

# Function to save the message to a file and update the latest_message variable
def save_message_to_file(message):
    global latest_message
    with open(message_log_file, "a") as file:
        file.write(message + "\n")
    latest_message = message

# Create the message log file if it doesn't exist
if not os.path.exists(message_log_file):
    open(message_log_file, "w").close()

# Create a simple HTTP server handler
class SimpleHTTPRequestHandler(http.server.BaseHTTPRequestHandler):
    def do_POST(self):
        content_length = int(self.headers["Content-Length"])
        message = self.rfile.read(content_length).decode("utf-8")
        save_message_to_file(message)

        print(message)
        self.send_response(200)
        self.end_headers()

    def do_GET(self):
        self.send_response(404)
        self.end_headers()

# Function to start the HTTP server

server_address = ("", 3000)
httpd = http.server.HTTPServer(server_address, SimpleHTTPRequestHandler)
print("Server started on port 3000")
httpd.serve_forever()

File: Bedrock Server Pilot/BSPChat Addon/http_server.mjs
import * as http from "http";
import * as fs from "fs";

const messageLogFile = "message_log.txt"; // File to save the messages

// Function to save the message to a file
function saveMessageToFile(message) {
  fs.appendFileSync(messageLogFile, `${message}\n`);
}

// Create the message log file if it doesn't exist
if (!fs.existsSync(messageLogFile)) {
  fs.writeFileSync(messageLogFile, "");
}

// Create a simple HTTP server
const server = http.createServer((req, res) => {
  if (req.method === "POST") {
    let data = "";
    req.on("data", chunk => {
      data += chunk;
    });
    req.on("end", () => {
      const message = data.toString();
      saveMessageToFile(message);
      console.log("Received message:", message);
      res.writeHead(200);
      res.end();
    });
  } else {
    res.writeHead(404);
    res.end();
  }
});

// Start the HTTP server
server.listen(3000, () => {
  console.log("Server started on port 3000");
});
